ui/src/components/SampleGrid/SampleGridTableItem.jsx

```diff
@@ -16,32 +15,24 @@ import containerStyles from '../../containers/SampleGridTableContainer.module.cs
 import TooltipTrigger from '../TooltipTrigger';
 import styles from './SampleGridTableItem.module.css';
 
-export class SampleGridTableItem extends React.Component {
-  static defaultProps = {
-    itemKey: '',
-    sampleData: {},
-    queueOrder: [],
-    selected: false,
-    current: false,
-    picked: false,
-    allowedDirections: [],
-    pickButtonOnClickHandler: undefined,
-  };
-
-  constructor(props) {
-    super(props);
-    this.pickButtonOnClick = this.pickButtonOnClick.bind(this);
-    this.handleItemClick = this.handleItemClick.bind(this);
-    this.sampleInformation = this.sampleInformation.bind(this);
-  }
-
-  pickButtonOnClick(e) {
-    if (this.props.pickButtonOnClickHandler) {
-      this.props.pickButtonOnClickHandler(e, this.props.sampleData.sampleID);
-    }
-  }
-
-  itemControls() {
+export default function SampleGridTableItem({
+  sampleData = {},
+  queueOrder = [],
+  current = false,
+  picked = false,
+  pickButtonOnClickHandler,
+  children,
+}) {
+  const pickButtonOnClick = useCallback(
+    (e) => {
+      if (pickButtonOnClickHandler) {
+        pickButtonOnClickHandler(e, sampleData.sampleID);
+      }
```

Comment:
I quite like this syntax personally, but it's a matter of taste:

```suggestion
      pickButtonOnClickHandler?.(e, sampleData.sampleID);
```

Reply:
I don't think you need to memoise this handler with `useCallback`. You can even just inline it in the JSX I think.